src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -24,10 +25,10 @@ public void addSession(MovieSession movieSession, User user) {
         newTicket.setUser(user);
         newTicket.setCinemaHall(movieSession.getCinemaHall());
         newTicket.setShowTime(movieSession.getShowTime());
-        newTicket.setMovie(movieSession.getMovie());
-
+        newTicket.setMovieSession(movieSession);
+        ticketDao.add(newTicket);
         ShoppingCart shoppingCart = shoppingCartDao.getByUser(user);
-        shoppingCart.getTickets().add(ticketDao.add(newTicket));
+        shoppingCart.setTickets(List.of(newTicket));
```

Comment:
If the user already had tickets in the cart they will be deleted and only 1 new ticket will remain